pricegraph/src/orderbook.rs

```diff
@@ -347,14 +347,18 @@ impl Orderbook {
     ///
     /// Note that the limit price is expressed as an exchange limit price, i.e.
     /// with implicitely included fees.
-    pub fn fill_order_at_price(&mut self, pair: TokenPair, actual_limit_price: f64) -> f64 {
+    pub fn fill_order_at_price(&mut self, pair: TokenPair, limit_price: f64) -> f64 {
         if !self.is_token_pair_valid(pair) {
             return 0.0;
         }
         self.update_projection_graph();
 
         let (sell, buy) = (node_index(pair.sell), node_index(pair.buy));
-        let effective_limit_price = actual_limit_price / FEE_FACTOR;
+        // NOTE: The limit price is expressed in `buy_amount / sell_amount` for
+        // the specified token pair, but since we are finding transitive orders
+        // in the opposite direction to match this order, calculate the maximum
+        // price that still respects the provided limit price.
+        let max_price = 1.0 / (limit_price * FEE_FACTOR);
```

Comment:
Happens to the best of us! I was actually wondering about this line before but neglected to comment. Good place to introduce a wee comment

Reply:
>  Good place to introduce a wee comment

Do you think the `// NOTE:` is not enough?